server/src/main/java/org/elasticsearch/common/util/BigIntArray.java

```diff
@@ -40,6 +42,21 @@ final class BigIntArray extends AbstractBigArray implements IntArray {
         }
     }
 
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (size > Integer.MAX_VALUE / Integer.BYTES) {
+            throw new IllegalArgumentException();
+        }
+        int intSize = (int) size;
+        out.writeVInt(intSize * 4);
+        for (int i = 0; i < pages.length - 1; i++) {
+            out.write(pages[i]);
+        }
+        int end = intSize % INT_PAGE_SIZE;
+        out.write(pages[pages.length - 1], 0, end * Integer.BYTES);
+        // NOCOMMIT endian-ness
```

Comment:
++ to your suggestion of adding a fallback for when the system byte order isn't LE if we have to.

But you do make a good point here, we aren't testing or supporting any BE platforms right now as far as I understand. Couldn't we just add a check to node startup that we're not on a BE platform and deal with the issue that way, saving some complexity?

Reply:
I've asked around and we don't support any big endian platforms at all. I think it's enough to fail to start if we are on such a system and to comment about it.